fe/fe-core/src/main/java/com/starrocks/sql/UpdatePlanner.java

```diff
@@ -84,8 +84,10 @@ public ExecPlan plan(UpdateStmt updateStmt, ConnectContext session) {
             long tableId = table.getId();
             List<Pair<Integer, ColumnDict>> globalDicts = Lists.newArrayList();
             for (Column column : table.getFullSchema()) {
-                if (updateStmt.usePartialUpdate() && !updateStmt.isAssignmentColumn(column.getName()) && !column.isKey()) {
-                    // When using partial update, skip columns which aren't key column and not be assign
+                if (updateStmt.usePartialUpdate() && !column.isMaterializedColumn() &&
```

Comment:
What's this meaning?
The generated column can be not written in the update DML?

Reply:
In partial update mode, generated column must be computed in OlapTableSink phase and can not be done in apply. The code here means that we SHOULD computed and write the generated column in the OlapTableSink phase. And it does not means that we could rewritten the  generated column using some specified value by user. e.g. update t set generated_column = value where pk = xxx will always throw an exception.